pkg/bootstrap/platform/gcp.go

```diff
@@ -235,7 +294,7 @@ func zoneToRegion(z string) (string, error) {
 // Locality returns the GCP-specific region and zone.
 func (e *gcpEnv) Locality() *core.Locality {
 	var l core.Locality
-	if metadata.OnGCE() {
+	if metadata.OnGCE() && isMetadataEndpointAccessibleFn() {
```

Comment:
should we just use `shouldFillMetadata`?

Reply:
I followed your suggestion of storing the result in the gcpEnv object.